src/ppl/nn/engines/cuda/cuda_common.h

```diff
@@ -20,6 +20,9 @@
 
 #if defined(__linux__)
 #include <sys/stat.h>
+
+#elif defined(__Win32) // defined(__Win32__) // defined(Win32)
+#include<iostream> 
 #endif
 
 #include <map>
```

Comment:
It is not specific to vs2019 on win32 or win64. Use #include <string> would make it work correctly on other OS and compilers such as clang on MacOSX.